rancher/rio
# Tag webhook support

Issue: https://github.com/rancher/rio/issues/542

First commit also adds shorthand `-n` to rio up.

Testing scenarios, I already tested all but need to go through again: 
* tags on or off, ie: tag-only vs commit-only webhooks
* using the tag-query and tag-exclude params
* tags on a non master branch still deploy - don't really need to test as long as tags work in general, this would just be testing github functionality
* Using CLI to try to make a service with tag and branch or tag without webhook should fail
* tags on a templated service should still work
* using riofile (see below)
* tags and filters on a riofile stack `rio up -n demo --build-tag=true --build-tag-include=^v --build-tag-exclude=alpha  https://github.com/....`

Testing instructions:
1. setup cluster
2. `rio secret add --github-webhook`
3. `rio run -p 8080 -n tag-demo --build-webhook-secret=githubtoken --build-tag=true --build-tag-include="^v" --build-tag-exclude="alpha" https://github.com/cbron/mybusybox`
4. push new tag or new commit, see that new pod is created for that service

For using local cluster, between steps 2 and 3 above do:
* expose your local machine
* `rio domain register external_endpoint rio-system:webhook`
* Do step 3
* update webhook to point to your external_endpoint

Example riofile: 
```
services:
  demo:
    build:
      repo: https://github.com/cbron/mybusybox
      branch: master
      webhookSecretName: githubtoken
      tag: true
      tagInclude: ^v
      tagExclude: rc
    ports:
    - 8080
```
